### PR TITLE
fix: adds autoMaxHeight prop to Modal

### LINKED ISSUE #1276: ️🛠 - Filter dropdown goes off-screen
### Task description

image

### Requirements

_No response_

### PR DESCRIPTION
**Pull request by jeeanribeiro**
## Summary

closes #1276 

### Platforms

-   **Desktop**
    -   [ ] MacOS
    -   [ ] Linux
    -   [x] Windows

## Checklist

-   [x] I have followed the contribution guidelines for this project
-   [x] I have performed a self-review of my own code
-   [ ] I have commented my code, particularly in hard-to-understand areas
-   [ ] I have added or modified tests that prove my changes work as intended
-   [x] I have verified that new and existing unit tests pass locally with my changes
-   [x] I have verified that my latest changes pass CI workflows for testing and linting
-   [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: packages/shared/src/components/modals/Modal.svelte [hunk 1]
```diff
@@ -39,38 +48,64 @@
         close()
     }
 
-    let isBlockedByTimeout = false
-    let show = false
+    async function updateMaxHeight(): Promise<void> {
+        await tick()
+        if (!show || !modal) return
```
Comment (Tuditi) [inline packages/shared/src/components/modals/Modal.svelte]: Please take care that we add return to a separate line
```suggestion
        if (!show || !modal) {
            return
        }
```

**Review by Tuditi — COMMENTED** [review]
Code looks good, but there is a lot of white space I added on the bottom if I enable the last options (see below). If it's a lot of work, we can make a separate polish issue out of this, so that the functionality works at least.

image

**Comment by jeeanribeiro** [comment]
> Code looks good, but there is a lot of white space I added on the bottom if I enable the last options (see below). If it's a lot of work, we can make a separate polish issue out of this, so that the functionality works at least.
> 
> image

That is the Dropdown component, we need to replace it by the UI kit SelectInput component, I will create an issue for that

**Comment by jeeanribeiro** [comment]
> Code looks good, but there is a lot of white space I added on the bottom if I enable the last options (see below). If it's a lot of work, we can make a separate polish issue out of this, so that the functionality works at least.
> 
> image

#1326 will fix this issue

**Review by Tuditi — APPROVED** [review]
LGTM! :fire:

**Pull request merged**